src/entity/SourcePostModeration.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post } from './posts';
+import { Source } from './Source';
+import { User } from './user';
+
+export enum SourcePostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SourcePostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
+
+  @Column({ type: 'text' })
+  status: SourcePostModerationStatus;
+
+  @Column({ type: 'text', nullable: true })
+  createdById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  createdBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatedById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'SET NULL',
+  })
+  moderatedBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatorMessage: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  rejectionReason: string | null;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column({ type: 'text', nullable: true })
+  postId: string | null;
+
+  @ManyToOne('Post', (post: Post) => post.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  post: Promise<Post>;
+
+  @Column({ type: 'text' })
+  type: string;
```

Comment:
Do we have some more specific type to apply? Something like `PostType`?

I see that in Post entity we are using this:

```ts
@Column({ default: PostType.Article })
type: PostType;
```

Maybe we can have a strict version only with squad-like posts.

Reply:
Good catch, it can be of type `PostType` because we want this to be generic not just for squads
```suggestion
  type: PostType;
```